pages/api/balance/[address].ts

```diff
@@ -0,0 +1,47 @@
+import { NextApiResponse, NextApiRequest } from 'next'
+import { getBCHBalance } from 'services/bchdService'
+import { RESPONSE_MESSAGES } from 'constants/index'
+import Cors from 'cors'
+
+const { ADDRESS_NOT_PROVIDED_400 } = RESPONSE_MESSAGES
+const cors = Cors({
+  methods: ['GET', 'HEAD'],
+})
+
+function runMiddleware(
+  req: NextApiRequest,
+  res: NextApiResponse,
+  fn: Function
+) {
+  return new Promise((resolve, reject) => {
+    fn(req, res, (result: any) => {
+      if (result instanceof Error) {
+        return reject(result)
+      }
+
+      return resolve(result)
+    })
+  })
+}
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void>  => {
+    await runMiddleware(req, res, cors)
+  if (req.method === 'GET') {
+    const address = req.query.address as string
```

Comment:
also, probably best to make this line `const address = parseAddress(req.query.address as string) ` so that the default parsing is executed